File: Nov2020/22Nov/MonkAndPowerOfTime.java
/*
Problem Name: Monk and Power of Time
Problem Link: https://www.hackerearth.com/practice/data-structures/queues/basics-of-queues/practice-problems/algorithm/monk-and-power-of-time-3a648bf0/
*/
import java.util.*;
class MonkAndPowerOfTime {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int arr[] = new int[n];
        Queue<Integer> q = new LinkedList<Integer>();
        for(int i=0;i<n;i++)
            q.add(in.nextInt());
        for(int i=0;i<n;i++)
            arr[i] = in.nextInt();
        int time = 0;
        for(int i=0;q.size()>0;){
            if(arr[i] == q.peek()){
                i++;
                q.remove();
            }
            else
                q.add(q.remove());
            time++;
        }
        System.out.println(time);
    }
}

File: Nov2020/22Nov/MonkAndPowerOfTime.cpp
/*
Problem Name: Monk and Power of Time
Problem Link: https://www.hackerearth.com/practice/data-structures/queues/basics-of-queues/practice-problems/algorithm/monk-and-power-of-time-3a648bf0/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int N;
    cin>>N;
    int val;
    vector<int> calling,ideal;
    for(int i=0;i<N;++i){
        cin>>val;
        calling.push_back(val);
    }
    for(int i=0;i<N;++i){
        cin>>val;
        ideal.push_back(val);
    }

    int ans=0,c=0,i=0;
    while(i<N){
        if(calling[c%N]==ideal[i]){
            i++;
            c++;
        }
        else{
            calling.push_back(calling[c]);
            calling.erase(calling.begin()+c);
            
        }    
        ans++;
    }
    cout<<ans;
    return 0;
}
